nifi-nar-bundles/nifi-hadoop-bundle/nifi-hdfs-processors/src/main/java/org/apache/nifi/processors/hadoop/AbstractHadoopProcessor.java

```diff
@@ -286,8 +310,10 @@ HdfsResources resetHDFSResources(String configResources, String dir, ProcessCont
                 }
             }
 
+            final Path workingDir = fs.getWorkingDirectory();
             getLogger().info("Initialized a new HDFS File System with working dir: {} default block size: {} default replication: {} config: {}",
-                    new Object[] { fs.getWorkingDirectory(), fs.getDefaultBlockSize(new Path(dir)), fs.getDefaultReplication(new Path(dir)), config.toString() });
+                    new Object[]{workingDir, fs.getDefaultBlockSize(workingDir), fs.getDefaultReplication(workingDir), config.toString()});
```

Comment:
@bbende I think it's more of a distant "maybe a possible problem in the future" and not an issue for now. thanks!